core-tests/src/test/java/org/mule/runtime/core/internal/processor/strategy/processor/PipelineMessageNotificationTestCase.java

```diff
@@ -287,13 +287,19 @@ public boolean matches(Notification argument) {
       EnrichedServerNotification notification = (EnrichedServerNotification) argument;
       Exception exception = notification.getException();
 
-      boolean result =
+      boolean result = true;
+      if (notification instanceof PipelineMessageNotification && exception instanceof MessagingException) {
```

Comment:
Why do you need to check for the instances?

Reply:
The one for the exception is not needed, but notification can be an ErrorHandlerNotification